streets_reader/src/split_ways.rs

```diff
@@ -202,24 +204,86 @@ pub fn split_up_roads(
                         road.dst_i
                     } else {
                         road.src_i
                     };
                     let i = streets.intersections.get_mut(&i).unwrap();
+                    // TODO Maybe we should do this later, as a consequence of TrafficInterruption
+                    // on incoming roads?
                     if !i.is_map_edge() {
                         i.control = IntersectionControl::Signalled;
                     }
+
+                    // Specify the explicit vehicle stop line
+                    if let Some((dist, _)) = road.reference_line.dist_along_of_point(pt.to_pt2d()) {
+                        let stop_line = if dir == Direction::Fwd {
+                            &mut road.stop_line_end
+                        } else {
+                            &mut road.stop_line_start
+                        };
+                        stop_line.vehicle_distance = Some(dist);
+                        stop_line.interruption = TrafficInterruption::Signal;
+                    }
+                    // TODO If dist_along_of_point fails, it's because we smoothed the line. This
+                    // is a great reason to instead just find the closest point on the line and
+                    // then the distance.
                 }
+                // TODO What should we do more generally with traffic signals on ways that don't
+                // specify a direction?
             }
         }
     }
     timer.stop("match traffic signals to intersections");
 
+    // Do the same for cycleway ASLs
+    for (pt, dir) in input.cycleway_stop_lines {
+        if let Some(road) = pt_to_road.get(&pt).and_then(|r| streets.roads.get_mut(r)) {
+            if let Some(dir) = dir {
+                if let Some((dist, _)) = road.reference_line.dist_along_of_point(pt.to_pt2d()) {
+                    let stop_line = if dir == Direction::Fwd {
+                        &mut road.stop_line_end
+                    } else {
+                        &mut road.stop_line_start
+                    };
+                    stop_line.bike_distance = Some(dist);
+
+                    // Inherit the interruption type from the intersection
+                    let i = if dir == Direction::Fwd {
+                        road.dst_i
+                    } else {
+                        road.src_i
+                    };
+                    if streets.intersections[&i].control == IntersectionControl::Signalled {
+                        stop_line.interruption = TrafficInterruption::Signal;
+                    }
+                }
+            }
+        }
+    }
+
+    for pt in input.signalized_crossings {
+        if let Some(road) = pt_to_road.get(&pt).and_then(|r| streets.roads.get_mut(r)) {
+            if let Some((dist, _)) = road.reference_line.dist_along_of_point(pt.to_pt2d()) {
+                // We don't know the direction. Arbitrarily snap to the start or end if it's within
```

Comment:
This is going to be wrong sometimes, see https://github.com/a-b-street/abstreet/issues/795#issuecomment-963688984

Reply:
If we are turning all footways into `Road`s will this be relavent any more? Each crossing will be an intersection in it's own right (which will often get merged into a signalised intersection).